kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,446 @@
+	}
+
+	@Test
+	public void getStringListShouldReturnOptionalStringListWithString() {
+		givenMapWith("testKey1", "testString1");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListOptIsCalledWith("testKey1");
+		thenNoExceptionsOccured();
+		thenOptionalResultEquals(Optional.of(Arrays.asList("testString1")));
+	}
+
+	@Test
+	public void getStringListShouldThrowWithNull() {
+		givenMapWith("testKeyNull", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListOptIsCalledWith("testKeyNull");
+		thenANoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void getStringListShouldThrowWithNoKey() {
+		givenMapWith("testKeyNull", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListOptIsCalledWith("");
+		thenANoSuchElementExceptionOccured();
+	}
+
+	/*
+	 * Given
+	 */
+
+	public void givenMapWith(String key, Object pair) {
+		properties.put(key, pair);
+	}
+
+	public void givenNetworkPropertiesBuiltWith(Map<String, Object> properties) {
+
+		try {
+			netProps = new NetworkProperties(properties);
+
+		} catch (NullPointerException e) {
+			this.hasNullPointExceptionBeenThrown = true;
+		} catch (NoSuchElementException e) {
+			this.hasNoSuchElementExceptionBeenThrown = true;
+		}
+	}
+
+	/*
+	 * When
+	 */
+
+	public void whenGetPropertiesIsCalled() {
+		this.resultMap = this.netProps.getProperties();
+	}
+
+	public void whenGetIsCalledWith(String key, Class<?> clazz) {
+
+		try {
+			if (clazz == String.class) {
+				this.stringResult = this.netProps.get(String.class, key, "");
+			} else if (clazz == Boolean.class) {
+				this.booleanResult = this.netProps.get(Boolean.class, key, "");
+			} else if (clazz == Short.class) {
+				this.shortResult = this.netProps.get(Short.class, key, "");
+			} else if (clazz == Integer.class) {
+				this.intResult = this.netProps.get(Integer.class, key, "");
+			} else if (clazz == Long.class) {
+				this.longResult = this.netProps.get(Long.class, key, "");
+			} else {
+				throw new NotSupported("Data type is not supported with this Test");
+			}
+
+		} catch (NullPointerException e) {
+			this.hasNullPointExceptionBeenThrown = true;
+		} catch (NoSuchElementException e) {
+			this.hasNoSuchElementExceptionBeenThrown = true;
+		}
+	}
+
+	public void whenGetOptIsCalledWith(String key, Class<?> clazz) {
+		try {
+			this.optResult = this.netProps.getOpt(clazz, key, "");
+		} catch (NullPointerException e) {
+			this.hasNullPointExceptionBeenThrown = true;
+		} catch (NoSuchElementException e) {
+			this.hasNoSuchElementExceptionBeenThrown = true;
+		}
+	}
+
+	public void whenGetStringListIsCalledWith(String key) {
+		try {
+			this.stringListResult = this.netProps.getStringList(key, "");
+
+		} catch (NullPointerException e) {
+			this.hasNullPointExceptionBeenThrown = true;
+		} catch (NoSuchElementException e) {
+			this.hasNoSuchElementExceptionBeenThrown = true;
+		}
+	}
+
+	public void whenGetStringListOptIsCalledWith(String key) {
```

Comment:
Please, follow the already established convention and use the correct method name:

```suggestion
	public void whenGetOptStringListIsCalledWith(String key) {
```